scipy/interpolate/tests/test_rbfinterp.py

```diff
@@ -0,0 +1,462 @@
+import pickle
+import numpy as np
+from numpy.testing import (
+    assert_, assert_allclose, assert_raises, assert_array_equal, assert_warns
+    )
+from scipy.spatial import cKDTree
+from scipy.interpolate.rbfinterp import (
+    _NAME_TO_MIN_DEGREE, _NAME_TO_FUNC, _SCALE_INVARIANT,
+    _vandermonde, _distance,
+    RBFInterpolator, KNearestRBFInterpolator
+    )
+
+
+def _1d_test_function(x):
+    # test function used in Wahba's "Spline Models for Observational Data
+    # domain ~= (0, 3), range ~= (-1.0, 0.2)"
+    x = x[:, 0]
+    y = 4.26*(np.exp(-x) - 4*np.exp(-2*x) + 3*np.exp(-3*x))
+    return y
+
+
+def _2d_test_function(x):
+    # Franke's test function
+    # domain ~= (0, 1) X (0, 1), range ~= (0.0, 1.2)"
+    x1, x2 = x[:, 0], x[:, 1]
+    term1 = 0.75 * np.exp(-(9*x1-2)**2/4 - (9*x2-2)**2/4)
+    term2 = 0.75 * np.exp(-(9*x1+1)**2/49 - (9*x2+1)/10)
+    term3 = 0.5 * np.exp(-(9*x1-7)**2/4 - (9*x2-3)**2/4)
+    term4 = -0.2 * np.exp(-(9*x1-4)**2 - (9*x2-7)**2)
+    y = term1 + term2 + term3 + term4
+    return y
+
+
+def _is_conditionally_positive_definite(kernel, m):
+    # Tests whether the kernel is conditionally positive definite of order m.
+    # See chapter 8 of Fasshauer's "Meshfree Approximation Methods with
+    # MATLAB".
+    np.random.seed(0)
+    nx = 10
+    ntests = 100
+    for ndim in [1, 2, 3, 4, 5]:
+        for _ in range(ntests):
+            x = np.random.normal(0.0, 1.0, (nx, ndim))
+            # add a small value to the diagonals in case the kernel _should_ be
+            # c.p.d but it is not due to numerical precision.
+            A = kernel(_distance(x, x)) + 1e-12*np.eye(nx)
+            P = _vandermonde(x, m - 1)
+            Q, R = np.linalg.qr(P, mode='complete')
+            # Q2 forms a basis spanning the space where P.T.dot(x) = 0. Project
+            # A onto this space, and then see if it is positive definite using
+            # the Cholesky decomposition. If not, then the kernel is not c.p.d
+            # of order m
+            Q2 = Q[:, P.shape[1]:]
+            B = Q2.T.dot(A).dot(Q2)
+            try:
+                np.linalg.cholesky(B)
+            except np.linalg.LinAlgError:
+                return False
+
+    return True
+
+
+def test_conditionally_positive_definite():
+    # Test if each kernel in _NAME_TO_FUNC is conditionally positive definite
+    # of order m, where m comes from _NAME_TO_MIN_DEGREE. This is a necessary
+    # condition for the smoothed RBF interpolant to be well-posed in general
+    for kernel_name, kernel_func in _NAME_TO_FUNC.items():
+        m = _NAME_TO_MIN_DEGREE.get(kernel_name, -1) + 1
+        assert_(_is_conditionally_positive_definite(kernel_func, m))
```

Comment:
I think we have a slight preference for plain `assert` now that we use `pytest`.

I suppose one could parametrize over `_NAME_TO_FUNC.items()` instead as well, although this often ends up in an annoying debate about test timings since we then end up with a separate test for each item being generated.